Replace AddSingleton with AddHostedService (#15089)
dotnet/docs: docs/architecture/microservices/multi-container-microservice-net-applications/background-tasks-with-ihostedservice.md

You can basically offload any of those actions to a background task based on IHostedService.

> The way you add one or multiple `IHostedServices` into your `WebHost` or `Host` is by registering them up through the [AddHostedService](https://docs.microsoft.com/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionhostedserviceextensions.addhostedservice) extension method in an ASP.NET Core `WebHost` (or in a `Host` in .NET Core 2.1 and above). Basically, you have to register the hosted services within the familiar `ConfigureServices()` method of the `Startup` class, as in the following code from a typical ASP.NET WebHost.

Reviewer: Let's use a special syntax that we have for API links instead.

```suggestion
The way you add one or multiple `IHostedServices` into your `WebHost` or `Host` is by registering them up through the <xref:Microsoft.Extensions.DependencyInjection.ServiceCollectionHostedServiceExtensions.AddHostedService%2A> extension method in an ASP.NET Core `WebHost` (or in a `Host` in .NET Core 2.1 and above). Basically, you have to register the hosted services within the familiar `ConfigureServices()` method of the `Startup` class, as in the following code from a typical ASP.NET WebHost.
```

> The way you add one or multiple `IHostedServices` into your `WebHost` or `Host` is by registering them up through the [AddHostedService](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionhostedserviceextensions.addhostedservice?view=aspnetcore-3.0) extension method in an ASP.NET Core `WebHost` (or in a `Host` in .NET Core 2.1 and above). Basically, you have to register the hosted services within the familiar `ConfigureServices()` method of the `Startup` class, as in the following code from a typical ASP.NET WebHost.

Reviewer 2: The link should not include the language or view specifiers:
```suggestion
The way you add one or multiple `IHostedServices` into your `WebHost` or `Host` is by registering them up through the [AddHostedService](https://docs.microsoft.com/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionhostedserviceextensions.addhostedservice) extension method in an ASP.NET Core `WebHost` (or in a `Host` in .NET Core 2.1 and above). Basically, you have to register the hosted services within the familiar `ConfigureServices()` method of the `Startup` class, as in the following code from a typical ASP.NET WebHost.
```

However @mairaw, what's the criteria to choose between a regular link or an xref link, and how do we get the correct xref we should link to?

Author: Thanks for catching that!

You can basically offload any of those actions to a background task based on IHostedService.

> The way you add one or multiple `IHostedServices` into your `WebHost` or `Host` is by registering them up through the [AddHostedService](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionhostedserviceextensions.addhostedservice) extension method in an ASP.NET Core `WebHost` (or in a `Host` in .NET Core 2.1 and above). Basically, you have to register the hosted services within the familiar `ConfigureServices()` method of the `Startup` class, as in the following code from a typical ASP.NET WebHost.

Reviewer 2: It's important to remove the localization segment from the path, so it adapts to the user's locale and doesn't interfere with automatic translation.

```suggestion
The way you add one or multiple `IHostedServices` into your `WebHost` or `Host` is by registering them up through the [AddHostedService](https://docs.microsoft.com/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionhostedserviceextensions.addhostedservice) extension method in an ASP.NET Core `WebHost` (or in a `Host` in .NET Core 2.1 and above). Basically, you have to register the hosted services within the familiar `ConfigureServices()` method of the `Startup` class, as in the following code from a typical ASP.NET WebHost.
```

Author: Good call, thank you
